test/validation/api/atomic/atomic.c

```diff
@@ -286,6 +286,54 @@ static void test_atomic_cas_inc_32(void)
 	}
 }
 
+static void test_atomic_cas_acq_inc_32(void)
+{
+	uint64_t i;
+	uint32_t old;
+	odp_atomic_u32_t *a32u = &global_mem->a32u;
+
+	odp_barrier_wait(&global_mem->global_barrier);
+
+	for (i = 0; i < CNT; i++) {
+		old = odp_atomic_load_u32(a32u);
+
+		while (odp_atomic_cas_acq_u32(a32u, &old, old + 1) == 0)
```

Comment:
CAS updates value of old when it did not succeed. Asserts could be added to check that value does not change on success, and changes on failure.